### PR TITLE
Version of trimZeros without temp strings

### PR DESCRIPTION
**Pull request by b3liever**
* removed usage of split

Also I think it should accept parameter ``decimalSep`` instead of guessing the separator.

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -2400,17 +2400,13 @@ proc trimZeros*(x: var string) {.noSideEffect.} =
...
-      x.setLen(x.len-1)
-    if spl.len > 0:
-      x &= "e" & spl[1]
+  if x.contains({'.', ','}):
+    let fE = x.find('e')
+    let last = if fE >= 0: fE-1 else: x.high
+    var first = last
+    while first >= 0 and x[first] == '0': dec(first)
+    if x[first] in {',', '.'}: dec(first)
+    x.delete(first+1, last)
```
Comment (juancarlospaco) [inline lib/pure/strutils.nim]: Spaces around operators.

File: lib/pure/strutils.nim [hunk 2]
```diff
@@ -2400,17 +2400,13 @@ proc trimZeros*(x: var string) {.noSideEffect.} =
...
-    if x[x.high] in [',', '.']:
-      x.setLen(x.len-1)
-    if spl.len > 0:
-      x &= "e" & spl[1]
+  if x.contains({'.', ','}):
+    let fE = x.find('e')
+    let last = if fE >= 0: fE-1 else: x.high
+    var first = last
+    while first >= 0 and x[first] == '0': dec(first)
+    if x[first] in {',', '.'}: dec(first)
```
Comment (b3liever) [inline lib/pure/strutils.nim]: original had no ``first >= 0`` checks, should I add one here too, or we don't care for bogus input that can crash it like ``"00e."``

**Comment by b3liever** [comment]
Rewrote algorithm myself, but if you know an other's language stdlib that has a same function let me know

File: lib/pure/strutils.nim (hunk 2 above)
Comment (b3liever) [reply lib/pure/strutils.nim]: won't crash now

**Comment by Araq** [comment]
I don't have an opinion. Seems fine the way it is. Unrelated CI failure, ready for merging as far as I'm concerned.

**Comment by b3liever** [comment]
ready

File: tests/stdlib/tstrutil.nim [hunk 3]
```diff
@@ -187,6 +187,20 @@ proc testRFind =
...
+  assert x == "1200"
+  x = "120.0"
+  x.trimZeros()
+  assert x == "120"
+  x = "0."
+  x.trimZeros()
+  assert x == "0"
+  x = "1.0e2"
+  x.trimZeros()
+  assert x == "1e2"
```
Comment (narimiran) [inline tests/stdlib/tstrutil.nim]: All these tests are either integers or floats of `x.0` format.

I would like to see more tests with different floats, just to be sure there's no some subtle bug hiding somewhere. Some floats I'd like to see tested (if you want, add more):

```
78.90
1.23e4
1.01
1.1001
0.0
0.01
1e0
```

Comment (b3liever) [reply tests/stdlib/tstrutil.nim]: I would add to the list ``70.85`` but I forgot

**Pull request merged**